### ISSUE TITLE
Terraform removes non-managed project metadata

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-google#39]
_This issue was originally opened by @cblecker as hashicorp/terraform#7636. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version

Terraform v0.6.16
### Affected Resource(s)
- google_compute_project_metadata
### Terraform Configuration Files

``` hcl
variable "region" {
  default = "us-east1"
}

variable "project_name" {
  description = "The ID of the Google Cloud project"
}

provider "google" {
  region      = "${var.region}"
  project     = "${var.project_name}"
  credentials = ""
}

resource "google_compute_project_metadata" "tf-testing-metadata" {
  metadata {
    tf-testing-1 = "test metadata variable 1 created by TF"
    tf-testing-2 = "test metadata variable 2 created by TF"
  }
}
```
### Debug Output

https://gist.github.com/cblecker/881acb52b2a3af9b194973ddd673fc08
### Panic Output

Not applicable
### Expected Behaviour

Any project metadata, including project-wide SSH keys created outside of Terraform should remain as part of the project should the Terraform-managed infrastructure be modified or destroyed.
### Actual Behaviour

The API call that Terraform uses to remove it's metadata from the project seems to clear all metadata from the project, including the very important sshKeys metadata key, which can result in loss of access to instances in the project that may not be managed by Terraform.
### Steps to Reproduce
1. Add non-Terraform managed metadata to a project: `gcloud compute project-info add-metadata --metadata testing-nontf-1=testing`
2. Verify metadata in the project: `gcloud compute project-info describe`
3. Add Terraform managed metadata to project: `terraform apply`
4. Verify metadata in the project: `gcloud compute project-info describe`
5. Destroy Terraform managed objects: `terraform destroy`
6. Verify metadata in the project: `gcloud compute project-info describe`
### Important Factoids

Not applicable
### References

Not applicable

**Comment by danawillow** [comment]
Fix in progress in #176

**Comment by cblecker** [comment]
That's awesome! Thanks @selmanj!

**Comment by rileykarson** [comment]
Fixed with #176.
